### PR TITLE
LMC multitask-SVGP models can output a single task per input.

### LINKED ISSUE #1285: [Question] Multi-output SVGP with irregular inputs?
Is there a way of training a multi-output variational GP model with irregular inputs, as in the  [Hadamard Multitask GP Regression](https://docs.gpytorch.ai/en/v1.2.0/examples/03_Multitask_Exact_GPs/Hadamard_Multitask_GP_Regression.html) example notebook?

Thanks,
Jimmy

### PR DESCRIPTION
**Pull request by gpleiss**
If one defines a ApproximateGP model with a LMCVariationalStrategy,
there are now two different options for return types:

1. Calling `model(x)` will return a `... x N x num_tasks`
MultitaskMultivariateNormal distribution
1. Calling `model(x, task_indices=i)` will return a `... x N`
MultivariateNormal distribution, where `i` corresponds to the selected
task index for each input.

[Closes #1285, #1433]
[Addresses #1743, #1765]

### CODE REVIEW

**Comment by gpleiss** [comment]
TODO:

- [ ] Update docs
- [ ] Update LMC tutorial

File: gpytorch/variational/lmc_variational_strategy.py [hunk 1]
```diff
@@ -120,28 +144,84 @@ def variational_params_initialized(self):
...
+        r"""
+        Computes the variational (or prior) distribution
+        :math:`q( \mathbf f \mid \mathbf X)` (or :math:`p( \mathbf f \mid \mathbf X)`).
+        There are two modes:
+
+        1.  Compute **all tasks** for all inputs.
+            If this is the case, the :attr:`task_indices` attribute should be None.
+            The return type will be a (... x N x num_tasks)
+            :class:`~gpytorch.distributions.MultitaskMultivariateNormal`.
+        2.  Compute **one tasks** per inputs.
```
Comment (jacobrgardner) [inline gpytorch/variational/lmc_variational_strategy.py:157]:
```suggestion
        2.  Compute **one task** per inputs.
```

File: gpytorch/variational/lmc_variational_strategy.py [hunk 2]
```diff
@@ -2,12 +2,34 @@
...
+    """
+    Given a list of indices for ... x N datapoints,
+      select the row from lmc_coefficient that corresponds to each datapoint
+
+    lmc_coefficients: torch.Tensor ... x num_latents x ... x num_tasks
+    indices: torch.Tesnor ... x N
+    """
+    batch_shape = _mul_broadcast_shape(lmc_coefficients.shape[:-1], indices.shape[:-1])
+
+    # We will use the left_t_interp helper to do the indexing
```
Comment (jacobrgardner) [inline gpytorch/variational/lmc_variational_strategy.py:24]:
```suggestion
    # We will use the left_interp helper to do the indexing
```

**Review by jacobrgardner — APPROVED** [review]
Looks good to me, and I actually need this PR for something :-)

**Comment by flcello** [comment]
As far as I understand it there is still some overhead in the standard `VariationalStrategy` because the conditional is still computed for all tasks and the relevant task is only selected afterwards in `IndependentMultitaskVariationalStrategy`. Is that right? Would it theoretically be possible to change that?

**Pull request merged**